src/Drivers/Payping/Payping.php

```diff
@@ -109,10 +109,10 @@ public function purchase()
             throw new PurchaseFailedException($message);
         }
 
-        $this->invoice->transactionId($this->invoice->getUuid());
+        $this->invoice->transactionId($body['code']);
 
         // return the transaction's id
-        return $this->invoice->getTransactionId();
+        return $this->invoice->getUuid();
```

Comment:
why you have changed this? purchase method MUST return transactionId. i can't merge this. it breaks the package abstractions.

you can access to invoice uuid using below syntax:

```php
// You can specify callbackUrl
Payment::callbackUrl('http://yoursite.com/verify')->purchase(
    $invoice, 
    function($driver, $transactionId) {
    	// We can store $transactionId in database.
        $invoice = $driver->getInvoice();
        $invoiceUuid = $invoice->getUuid();
	}
);
```

you have access to invoice and driver as you see.

see line 88 on the below link

https://github.com/shetabit/multipay/blob/master/src/Abstracts/Driver.php

Reply:
so please help me to figure out how we can get $transaction_id here 
```
$receipt = $payment->amount(1000)->transactionId($transaction_id)->verify();
```

```
// At the top of the file.
use Shetabit\Multipay\Payment;
use Shetabit\Multipay\Exceptions\InvalidPaymentException;
...

// load the config file from your project
$paymentConfig = require('path/to/payment.php');

$payment = new Payment($paymentConfig);


// You need to verify the payment to ensure the invoice has been paid successfully.
// We use transaction id to verify payments
// It is a good practice to add invoice amount as well.
try {
	$receipt = $payment->amount(1000)->transactionId($transaction_id)->verify();

    // You can show payment referenceId to the user.
    echo $receipt->getReferenceId();

    ...
} catch (InvalidPaymentException $exception) {
    /**
    	when payment is not verified, it will throw an exception.
    	We can catch the exception to handle invalid payments.
    	getMessage method, returns a suitable message that can be used in user interface.
    **/
    echo $exception->getMessage();
}
```